darshan-runtime/doc/darshan-runtime.txt

```diff
@@ -478,6 +446,147 @@ git submodule update --init
+
+For example, the following settings would disable the generation of Darshan logs for `git` and `ls`.
+
+----
+export DARSHAN_APP_EXCLUDE="git,ls"
+----
+
+Darshan also allows for instrumentation to be disabled on specific application ranks.
+Users may set an exclusion list as well as an overriding inclusion list. Independent
+ranks or rank ranges may be specified, with ranges in the formatted like "start:end"
+(if start or end are not specified, the first or last rank is assumed, respectively).
+Note that the Darshan library will still run on all processes of an application,
+this setting just controls whether specific ranks are storing instrumentation data.
+
+* DARSHAN_RANK_EXCLUDE: specifies a list of comma-separated rank specifiers which should not be instrumented
+* DARSHAN_RANK_INCLUDE: specifies a list of comma-separated rank specifiers which should be instrumented (i.e., overrides DARSHAN_RANK_EXCLUDE specifiers)
+
+For example, the following settings would disable Darshan instrumentation for all
+application ranks except for 0-9:
+
+----
+export DARSHAN_RANK_EXCLUDE="0:"  # disable for all ranks first
+export DARSHAN_RANK_INCLUDE="0:9" # selectively enable a rank range
+----
+
+Users can also configure a couple of filters for DXT trace data which
+control whether trace records are kept or discarded at application shutdown
+time. This is useful for focusing DXT instrumentation data on file access
+patterns that are of particular interest from an I/O performance perspective:
+
+* DARSHAN_DXT_SMALL_IO_TRIGGER: specifies a floating point percentage (i.e., ".8" would be 80%) of small I/O operations (defined as operations smaller than 10 KiB). Traces are only retained for files that exhibit a percentage of small I/O operations greater than this threshold.
+* DARSHAN_DXT_UNALIGNED_IO_TRIGGER: specifies a floating point percentage (i.e., ".8" would be 80%) of unaligned I/O operations (not aligned to file alignment value determined by Darshan). Traces are only retained for files that exhibit a percentage of unaligned I/O operations greater than this threshold.
+
+For example, the following settings would ensure traces are only kept for files
+that were accessed using small I/O operations 20+% of the time:
+
+----
+export DARSHAN_DXT_SMALL_IO_TRIGGER=".2"
+----
+
+=== Darshan library config file
+
+Darshan also provides the ability for users to specify a configuration file to
+load library settings from. It is important to note that for settings that are
+specified in both the environment and via a config file, the environment
+settings will take precedence.
+
+Users may specify the config file location using the `DARSHAN_CONFIG_PATH`
+environment variable. The config file allows users to specify the following
+settings, one per-line, with setting names and values separated by any whitespace.
+Many of the settings directly match settings defined in section 
```

Comment:
That was kind of the intent, there are only a handful of exceptions right now, maybe you can comment on whether these exceptions make sense:

- `MAX_RECORDS` and `NAME_EXCLUDE`/`NAME_INCLUDE`: Not supported as env variable variants because these settings require a value and a list of modules to apply them to, which is complicated to express as an env var. You'd either have to come up with some wonky formatting in a single string, or you'd have to define independent env vars for each module which I don't think we want.
- `DXT_ENABLE_IO_TRACE`: Not supported in config file for a couple of reasons. 1) `MODULE_ENABLE` settings can accomplish the same thing, so decided it might be simplest to only honor `DXT_ENABLE_IO_TRACE` as an env var as a convenience to users who have gotten used to it that way. 2) This env var isn't prefixed with `DARSHAN`, which makes it an oddball to try to support in the config file in this 1-1 way (chop off the `DARSHAN_` part of the env var to get the config file setting).
- `DARSHAN_EXCLUDE_DIRS`: Not supported in config file for similar reasons to `DXT_ENABLE_IO_TRACE`, in that the config file has a new way of doing the same thing via `NAME_EXCLUDE`. We could honor this in the config file, I just think there's a tradeoff to being consistent and not advertising multiple ways for users to do the same thing (while still honoring old env vars that users may have gotten used to).

So, first is because of complexity, and the latter 2 are related to deprecation.

Beyond that, there are some smaller examples of discrepancies with the following only available as env vars:
- `DARSHAN_DUMP_CONFIG`:
- `DARSHAN_INTERNAL_TIMING` 
- `DARSHAN_DISABLE_SHARED_REDUCTION`
- `DARSHAN_DISABLE`
- `DARSHAN_LOGNAME`

I suppose we could offer config file variants for the first 3 if we wanted, they just kind of struck me as small things that could just stick to the environment only. The last 2 don't really seem appropriate for a config file.

Reply:
I agree with the comments in general though, I'll take another pass over this part of the documentation and try to unify as much of this as possible instead of repeating a lot of the things in 2 different sections. I'll wait and see your feedback on discrepancies before hacking up any changes to the code though.

Reply:
Okay, I think these changes are reflected in my recent commits. It'd probably be most helpful if you can scan the updated docs to see if the new section and table I came up with are sensible. There are some changes to runtime code to support more settings in the config file approach (specifically, `DISABLE_SHARED_REDUCTION`, `DUMP_CONFIG`, and `INTERNAL_TIMING`).

I also tried to make sure to be more consistent in how users can set settings depending on whether they use env vars or a config file. Specifically, I made sure to be consistent that users can provide CSVs of regular expressions for all settings, regardless of whether they are using env vars or the config file -- obviously, that is a necessity for env vars (you can't set them multiple times like you can repeatedly specify config file settings), but I wasn't supporting that in the config file approach. Now, the config file allows you to specify these parameters using multiple lines, or using comma-separated values, if that makes sense.